frontend/App.js

```diff
@@ -49,6 +58,29 @@ export default class App extends React.Component {
     );
   }
 
+  async allowPermission(type) {
```

Comment:
We've changed this logic. Should just be `askAsync` now. If the permission exists, it won't ask again.

Reply:
I am going to have to push back on this, I couldn't get it to fire on iOS doing you suggested method. Take a look whenever you have time. 

I am using this as a reusable function `await this.allowPermission('CAMERA')`
Which then uses the suggested const: `Permissions.CAMERA` depending on why type of permission is requested

Reply:
It seems to work as I had expected: https://snack.expo.io/@bacon/image-upload-s3 
Am I missing something?

Reply:
Oh my word, I am so sorry, I apologize!

I was not destructuring when i tried to do your suggested solution and my helper function was handling the 'granted' to be a true or false. That was 100% my bad, so so sorry. I've fixed it and pushed to this PR.